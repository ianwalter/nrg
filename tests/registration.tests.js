const { test } = require('@ianwalter/bff')
const app = require('../examples/accounts')

const firstName = 'Bilbo'
const lastName = 'Baggins'
const email = 'bilbo@example.com'
const password = '13eip3mlsdf0123mklqslk'

test('Registration required validation', async ({ expect }) => {
  // Email required.
  let payload = { firstName, lastName, password }
  let response = await app.test('/registration').post(payload)
  expect(response.status).toBe(400)
  expect(response.body).toMatchSnapshot()

  // Password required.
  payload = { firstName, lastName, email }
  response = await app.test('/registration').post(payload)
  expect(response.status).toBe(400)
  expect(response.body).toMatchSnapshot()

  // First name required.
  payload = { firstName: '', lastName, email, password }
  response = await app.test('/registration').post(payload)
  expect(response.status).toBe(400)
  expect(response.body).toMatchSnapshot()

  // Last name required.
  payload = { firstName, lastName: null, email, password }
  response = await app.test('/registration').post(payload)
  expect(response.status).toBe(400)
  expect(response.body).toMatchSnapshot()
})

test('Registration with weak password', async ({ expect }) => {
  const payload = { firstName, lastName, email, password: 'abc123' }
  const response = await app.test('/registration').post(payload)
  expect(response.status).toBe(400)
  expect(response.body).toMatchSnapshot()
})

test('Registration with invalid email', async ({ expect }) => {
  const payload = { firstName, lastName, email: 'bilbo@example,com', password }
  const response = await app.test('/registration').post(payload)
  expect(response.status).toBe(400)
  expect(response.body).toMatchSnapshot()
})

test('Registration with valid data', async ({ expect }) => {
  const payload = { firstName, lastName, email, password }
  const response = await app.test('/registration').post(payload)
  expect(response.status).toBe(200)
  expect(response.body).toMatchSnapshot()
})
