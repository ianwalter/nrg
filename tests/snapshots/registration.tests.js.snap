// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Registration required validation 1`] = `
Object {
  "feedback": Object {
    "email": Array [
      "A valid email is required.",
    ],
  },
  "message": "Validation Error",
}
`;

exports[`Registration required validation 2`] = `
Object {
  "feedback": Object {
    "password": Array [
      "A valid password is required.",
    ],
  },
  "message": "Validation Error",
}
`;

exports[`Registration required validation 3`] = `
Object {
  "feedback": Object {
    "firstName": Array [
      "A valid first name is required.",
    ],
  },
  "message": "Validation Error",
}
`;

exports[`Registration required validation 4`] = `
Object {
  "feedback": Object {
    "lastName": Array [
      "A valid last name is required.",
    ],
  },
  "message": "Validation Error",
}
`;

exports[`Registration with invalid email 1`] = `
Object {
  "feedback": Object {
    "email": Array [
      "A valid email is required.",
    ],
  },
  "message": "Validation Error",
}
`;

exports[`Registration with valid data 1`] = `
Object {
  "message": "Email Verification request submitted successfully",
}
`;

exports[`Registration with weak password 1`] = `
Object {
  "feedback": Object {
    "password": Array [
      "A valid password is required.",
    ],
  },
  "message": "Validation Error",
}
`;
